channel.go

```diff
@@ -62,6 +62,10 @@ type ChannelOptions struct {
 	// unstable API - breaking changes are likely.
 	RelayHost RelayHost
 
+	// OnPeerStatusChanged receives notifications when a peer has become
+	// available or unavailable.
```

Comment:
I think we should expand the comment here with more details -- I this method should be called for more events than a peer becoming "available" or "unavailable". Basically any change to the state of the peer (new connections, closed connections) should trigger this right?